addons/mail/static/src/discuss/core/public_web/discuss_channel_model_patch.js

```diff
@@ -93,25 +84,6 @@ const discussChannelPatch = {
             { description }
         );
     },
-    get allowedToLeaveChannelTypes() {
-        return ["channel", "group"];
-    },
-    get allowedToUnpinChannelTypes() {
-        return ["chat"];
-    },
```

Comment:
Note: removed as all channel types can now be potentially hidden (`canHide` can refine the condition).